decompiler/frontend/binaryninja/frontend.py

```diff
@@ -53,19 +53,21 @@ def from_raw(cls, view: BinaryView):
     def create_task(self, function: Union[str, Function], options: Options) -> DecompilerTask:
         """Create a task from the given function identifier."""
         debug_mode = options.getboolean("pipeline.debug", fallback=False)
-        if not isinstance(function, Function):
-            function = self._find_function(function)
-        return_type, params = self._extract_return_type_and_params(function)
+        if isinstance(function, Function):
+            function_obj = function
+        elif (function_obj := self._get_function_from_str(function)) is None:
+            raise ValueError(f"Could not resolve function symbol {function}")
+        return_type, params = self._extract_return_type_and_params(function_obj)
```

Comment:
I feel like there might be a class hiding here (e.g. `FunctionObject.from_string` and obj.return_type`)
I feel this function is getting a little long. What about extracting the if-construct? 

It would also help if we did not allow strings here e.g. resolved string names somewhere else

Reply:
Implemented this class